File: 02-resilience-and-caching/backend/node-server/src/services/rateLimiter.js
import redis from '../redis.js';

// Rate limiting por ventana fija: INCR + EXPIRE en la 1ª petición.
async function check() {
  const n = await redis.incr('ratelimit:global');
  if (n === 1) {
    await redis.expire('ratelimit:global', 10);  // 1ª petición abre la ventana
  }
  return { permitted: n <= 30, n, limited: 30, window: 10 };
}

export default { check };

File: 02-resilience-and-caching/backend/python-server/src/services/rate_limiter.py
from redis_client import get_redis

RATE_KEY = "ratelimit:global"
VENTANA = 10  # segundos por ventana
LIMITE = 30   # peticiones máximas por ventana


# Rate limiting por ventana fija: INCR + EXPIRE en la 1ª petición.
def check():
    r = get_redis()
    n = r.incr(RATE_KEY)
    if n == 1:
        r.expire(RATE_KEY, VENTANA)  # 1ª petición abre la ventana
    return {"permitted": n <= LIMITE, "n": n, "limited": LIMITE, "window": VENTANA}